cds-snc/platform-forms-client
# Character limit modal

# Summary | Résumé

This PR adds the numeric input field, and adds a "character limit" setting to the modal for text input fields. It also makes the SVG sizes more consistent in the field selector.

## Numeric input field 

The numeric input field was in figma as a setting in the modal, but it actually makes more sense as a field type. Unlike "required", which can be applied to any field, the numeric entry only really makes sense in normal text fields, which makes it a specialized version of a text field. In the form itself, it shows up as an `<input type="number" />`, which means that it only permits numbers, as well as `-`, `+`, `.`, and `e` characters. 

| in Figma                                          | as its own field type                                        |
| ----------------------------------------------- | -------------------------------------------- |
| In Figma, this was an option in the modal | It's been implemented as a specialized field |
| <img width="732" alt="Screen Shot 2022-09-26 at 09 52 59" src="https://user-images.githubusercontent.com/2454380/192295352-18b61623-57a5-49e2-a49d-a855b5351c2d.png"> | <img width="826" alt="Screen Shot 2022-09-24 at 19 01 53" src="https://user-images.githubusercontent.com/2454380/192295349-9afdff03-b1e4-48eb-9ac1-1742ba92a703.png"> |


## "Character limit" option in modal 

Added a character limit option to the modal with a custom implementation of the `<input type="number" />` field that doesn't allow special characters (eg, no `e`). It only exists for the basic `textField` at the minute ("Short answer"), although we can include it in other fields in the future if we like. Here's a gif of it being set.

| Character limit                                          | 
| ----------------------------------------------- | 
|  ![Screen Recording 2022-09-24 at 19 03 24](https://user-images.githubusercontent.com/2454380/192296420-9d48f542-a073-4422-bf9f-da9e3a16d940.gif) | 

## How to test this change

Add a numeric field to the form and add a character limit to a Short answer, and then test them in "preview" mode. Once you chance the type of your Short answer question, the character limit should vanish.
